src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,327 @@
+            {
+                worker.getInventoryCitizen().setInventorySlotContents(i, null);
+                continue;
+            }
+
+            if(stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType)== null)
+            {
+                worker.setItemStackToSlot(((ItemArmor) stack.getItem()).armorType, stack);
+            }
+        }
+    }
+
+    /**
+     * Chooses a target from the list.
+     * @return the next state.
+     */
+    protected AIState getTarget()
+    {
+        if(entityList.isEmpty())
+        {
+            return AIState.GUARD_PATROL;
+        }
+
+        if(entityList.get(0) instanceof EntityPlayer)
+        {
+            if(worker.getColony() != null && worker.getColony().getPermissions().hasPermission((EntityPlayer) entityList.get(0), Permissions.Action.GUARDS_ATTACK))
+            {
+                targetEntity = (EntityLivingBase) entityList.get(0);
+                worker.getNavigator().clearPathEntity();
+                return AIState.GUARD_HUNT_DOWN_TARGET;
+            }
+            entityList.remove(0);
+            setDelay(BASE_DELAY);
+            return AIState.GUARD_GET_TARGET;
+        }
+        else if (!worker.getEntitySenses().canSee(entityList.get(0)) || !(entityList.get(0)).isEntityAlive())
+        {
+            entityList.remove(0);
+            setDelay(BASE_DELAY);
+            return AIState.GUARD_GET_TARGET;
+        }
+        else
+        {
+            worker.getNavigator().clearPathEntity();
+            targetEntity = (EntityLivingBase) entityList.get(0);
+            return AIState.GUARD_HUNT_DOWN_TARGET;
+        }
+    }
+
+    /**
+     * Searches for the next taget.
+     * @return the next AIState.
+     */
+    protected AIState searchTarget()
+    {
+        entityList = this.worker.worldObj.getEntitiesWithinAABB(EntityMob.class, this.getTargetableArea(currentSearchDistance));
+        entityList.addAll(this.worker.worldObj.getEntitiesWithinAABB(EntitySlime.class, this.getTargetableArea(currentSearchDistance)));
+        entityList.addAll(this.worker.worldObj.getEntitiesWithinAABB(EntityPlayer.class, this.getTargetableArea(currentSearchDistance)));
+
+        setDelay(BASE_DELAY);
+        if(entityList.isEmpty())
+        {
+            if(currentSearchDistance < getMaxVision())
+            {
+                currentSearchDistance += START_SEARCH_DISTANCE;
+            }
+            else
+            {
+                currentSearchDistance = START_SEARCH_DISTANCE;
+                return AIState.GUARD_PATROL;
+            }
+
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        return AIState.GUARD_GET_TARGET;
+    }
+
+    /**
+     * Getter for the vision or attack distance.
+     * @return the max vision.
+     */
+    private double getMaxVision()
+    {
+        BuildingGuardTower guardTower = (BuildingGuardTower) worker.getWorkBuilding();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'guardTower' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)